lib/manageiq/active_record_connector.rb

```diff
@@ -0,0 +1,60 @@
+require "active_record"
+unless defined? DatabaseConfigurationPatch
+  require_relative "../patches/database_configuration_patch.rb"
+end
+
+module ManageIQ
+  class ActiveRecordConnector
+    # Helper class for getting the database config without needing Rails
+    #
+    # Makes use of the database_configuration_patch for the most part for
+    # grabbing this from the yaml file in config/database.yml
+    class ConnectionConfig
+      # Set a default method for handling super call in
+      # lib/patches/database_configuration_patch.rb
+      def self.database_configuration
+        if ENV["DATABASE_URL"]
+          {}
+        end
+      end
+
+      class << self
+        prepend DatabaseConfigurationPatch
+      end
+
+      def self.[](env)
+        database_configuration[env]
+      end
+    end
+
+    # Returns true if an ActiveRecord::Base.connection_config is present,
+    # otherwise returns false
+    def self.connection_exists?
+      begin
+        true if ActiveRecord::Base.connection_config
+      rescue ActiveRecord::ConnectionNotEstablished
+        false
+      end
+    end
+
+    # Establishes a connection if one doesn't already exists.  If a block is
+    # passed in, run the code within the block, and then remove the
+    # connections that have been made by this method.
+    def self.establish_connection_if_needed(db_config, log_path)
+      existing_connection = connection_exists?
+
+      unless existing_connection
+        ActiveRecord::Base.logger = Logger.new(log_path)
```

Comment:
Quick follow up to my thought with that:

I assume that `lib/vmdb/logging.rb` would be in charge of the handling any common logging stuff in the future, and that would get initialized prior to making the database connection.  So if that is then the case, we can then make the `.establish_connection_if_needed` call simpler by not requiring the `log_path`, but gracefully set a default when a `ActiveRecord::Base.logger` isn't set and a `log_path` isn't present.